controllers/space/space_controller_test.go

```diff
@@ -0,0 +1,796 @@
+	"github.com/codeready-toolchain/host-operator/controllers/space"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/pkg/cluster"
+	. "github.com/codeready-toolchain/host-operator/test"
+	tiertest "github.com/codeready-toolchain/host-operator/test/nstemplatetier"
+	spacetest "github.com/codeready-toolchain/host-operator/test/space"
+	commoncluster "github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	nstemplatetsettest "github.com/codeready-toolchain/toolchain-common/pkg/test/nstemplateset"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestReconciler(t *testing.T) {
+
+	// given
+	logf.SetLogger(zap.New(zap.UseDevMode(true)))
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+	basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates)
+
+	t.Run("create", func(t *testing.T) {
+
+		t.Run("success", func(t *testing.T) {
+			// given
+			space := spacetest.NewSpace("oddity", basicTier.Name, spacetest.WithTargetCluster("member-1"))
+			hostClient := test.NewFakeClient(t, space, basicTier)
+			member1 := NewMemberCluster(t, "member-1", corev1.ConditionTrue)
+			member2 := NewMemberCluster(t, "member-2", corev1.ConditionTrue)
+			ctrl := newReconciler(hostClient, member1, member2)
+
+			// when
+			res, err := ctrl.Reconcile(context.TODO(), reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: space.Namespace,
+					Name:      space.Name,
+				},
+			})
+
+			// then
+			require.NoError(t, err)
+			assert.False(t, res.Requeue)
+			nsTmplSet := nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, "oddity", member1.Client).
+				Exists().
+				Get()
+			spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
+				Exists().
+				HasStatusTargetCluster("member-1").
+				HasConditions(spacetest.Provisioning()).
+				HasFinalizer(toolchainv1alpha1.FinalizerName)
+
+			t.Run("requeue while NSTemplateSet is not ready", func(t *testing.T) {
+				// given another round of requeue without while NSTemplateSet is *not ready*
+				nsTmplSet.Status.Conditions = []toolchainv1alpha1.Condition{
+					{
+						Type:   toolchainv1alpha1.ConditionReady,
+						Status: corev1.ConditionFalse,
+						Reason: toolchainv1alpha1.NSTemplateSetProvisioningReason,
+					},
+				}
+				err := member1.Client.Update(context.TODO(), nsTmplSet)
+				require.NoError(t, err)
+				ctrl := newReconciler(hostClient, member1, member2)
```

Comment:
I thought that we had issues in the past where a unit test reused a controller, which resulted in errors? or am I missing something?

Reply:
I'm not aware of anything like that. In the subtests you use the same objects, the same clients, so I would expect that you should use also the same Controller object.
if it wasn't a subtests, then it would be a different story - we shouldn't reuse the same controller between other tests that are at the same level.

Reply:
I probably got confused with this [discussion on our Slack channel](https://coreos.slack.com/archives/CHK0J6HT6/p1633509468123100?thread_ts=1633503180.103700&cid=CHK0J6HT6)

Reply:
removed in 44a3b31